sway-ast/src/keywords.rs

```diff
@@ -232,3 +233,99 @@ define_token!(DoublePipeToken, "`||`", [Pipe, Pipe], [Pipe]);
 define_token!(UnderscoreToken, "`_`", [Underscore], [Underscore]);
 define_token!(HashToken, "`#`", [Sharp], []);
 define_token!(HashBangToken, "`#!`", [Sharp, Bang], []);
+
+pub trait OpenDelimiterToken: Spanned + Sized {
+    /// Creates the keyword from the given `span`.
+    fn new(span: Span) -> Self;
+
+    /// Returns an identifier for this keyword.
+    fn ident(&self) -> Ident;
+
+    /// The sequence of punctuations that make up the token.
+    const DELIMITER_KIND: &'static [OpeningDelimiter];
+}
+macro_rules! define_opening_delimiter_token (
```

Comment:
Again, my intuition is that we don't need to have unique traits for closing or opening delimeters and we can find a sollution that is generic over either type. Is it possible to rename this to `define_delimiter_token` and make it work for both types?